platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/metrics/BukkitMetricsManager.java

```diff
@@ -0,0 +1,76 @@
+import org.bstats.charts.SimplePie;
+import org.bukkit.Bukkit;
+import org.bukkit.plugin.RegisteredServiceProvider;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class BukkitMetricsManager {
+
+    private final TreasuryBukkit plugin;
+    public BukkitMetricsManager(final @NotNull TreasuryBukkit plugin) { this.plugin = plugin; }
+
+    Metrics metrics;
+
+    public void load() {
+        metrics = new Metrics(plugin, 12927);
+
+        metrics.addCustomChart(new SimplePie("economy-provider-name", () -> {
+            final RegisteredServiceProvider<EconomyProvider> economyProvider = getEconomyProviderRegistration();
+            return economyProvider == null ? "None" : economyProvider.getPlugin().getName();
+        }));
+
+        metrics.addCustomChart(new SimplePie("economy-provider-supports-negative-balances", () -> {
+            final RegisteredServiceProvider<EconomyProvider> economyProvider = getEconomyProviderRegistration();
+            return economyProvider == null ? null :
+                    economyProvider.getProvider().getSupportedOptionalEconomyApiFeatures()
+                            .contains(OptionalEconomyApiFeature.NEGATIVE_BALANCES) + "";
+        }));
+
+        metrics.addCustomChart(new SimplePie("economy-provider-supports-bukkit-transaction-events", () -> {
+            final RegisteredServiceProvider<EconomyProvider> economyProvider = getEconomyProviderRegistration();
+            return economyProvider == null ? null :
+                    economyProvider.getProvider().getSupportedOptionalEconomyApiFeatures()
+                            .contains(OptionalEconomyApiFeature.BUKKIT_TRANSACTION_EVENTS) + "";
+        }));
+
+        metrics.addCustomChart(new SimplePie("economy-treasury-api-version", () -> {
+            //noinspection deprecation
+            return EconomyAPIVersion.getCurrentAPIVersion().toString();
+        }));
+
+        metrics.addCustomChart(new SimplePie("economy-provider-api-version", () -> {
+            final RegisteredServiceProvider<EconomyProvider> economyProvider = getEconomyProviderRegistration();
+            return economyProvider == null ? null :
+                    economyProvider.getProvider().getSupportedAPIVersion().toString();
+        }));
+
+        metrics.addCustomChart(new SimplePie("plugin-update-checking-enabled", () ->
+                plugin.getTreasuryPlugin().getSettings().checkForUpdates() + ""));
+
+        metrics.addCustomChart(new SimplePie("economy-provider-supports-negative-balances", () -> {
+            final RegisteredServiceProvider<EconomyProvider> economyProvider = getEconomyProviderRegistration();
+            if(economyProvider == null) { return null; }
+
+            final int size = economyProvider.getProvider().getCurrencies().size();
+
+            if(size >= 10) {
+                return "10+";
+            } else {
+                return size + "";
+            }
```

Comment:
Why not just return the amount?

Reply:
maybe because he doesn't want the chart to be dirty?

Reply:
We need the chart to only show useful data. If one person decides to have 73 currencies then we really don't care about it, anything more than 10 should be in its own group because it translates to 'a ton of currencies' :smiley: